crux-agent/app/api/routers/jobs.py

```
@@ -350,6 +350,56 @@ async def cancel_job(
+            "job_id": job_id,
+            "status": "purged",
+            "message": "Job and all associated data removed",
+        }
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error purging job: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to purge job: {str(e)}",
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)